lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2136,27 +2147,39 @@ def _enable_ssh_on_windows(self, node: Node) -> None:
         log.dump_json(logging.DEBUG, result)
 
     def _get_vhd_os_disk_size(self, blob_url: str) -> int:
-        vhd_blob = BlobClient.from_blob_url(
-            blob_url=blob_url,
+        matched = STORAGE_CONTAINER_BLOB_PATTERN.match(blob_url)
```

Comment:
The from blob url should wrap this logic already. When does not it work?

Reply:
any feedback?

Reply:
Can you paste the error again? I want to double check if from_blob_url work with this scenario or not.

Reply:
deployment failed. HttpResponseError: Operation returned an invalid status 'This request is not authorized to perform this operation using this permission.'
ErrorCode:AuthorizationPermissionMismatch